### PR TITLE
Connection pool

### PR DESCRIPTION
**Pull request by mpollmeier**
Using the connection pool similar to what the TP2 driver did.

### CODE REVIEW

File: driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraphFactory.java [hunk 1]
```diff
@@ -54,4 +100,40 @@ protected Configuration getConfiguration(boolean create, boolean open, boolean t
...
+        } else if (open) db.open(user, password);
+
+        return db;
+    }
+
+    /**
+     * Setting up the factory to use database pool instead of creation a new instance of database connection each time.
+     */
+    public OrientGraphFactory setupPool(final int max) {
+        pool = new OPartitionedDatabasePool(url, user, password, max).setAutoCreate(true);
```
Comment (velo) [inline driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraphFactory.java]: Would be also nice, to make this option available using configuration...

That way, or tests can randomly include it and so some tests would run under pool and others w/o pool.

File: driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java [hunk 2]
```diff
@@ -69,26 +70,30 @@
 
     protected final ODatabaseDocumentTx database;
     protected final Features features;
-    protected final String url;
-    private final Configuration configuration;
-
-    public static OrientGraph open(final Configuration configuration) {
```
Comment (velo) [inline driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java]: This does cause
````
java.lang.RuntimeException: GraphFactory can only instantiate Graph implementations from classes that have a static open() method that takes a single Apache Commons Configuration argument - [class org.apache.tinkerpop.gremlin.orientdb.OrientGraph] does not seem to have one
	at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:80)
	at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:70)
	at org.apache.tinkerpop.gremlin.GraphProvider.openTestGraph(GraphProvider.java:130)
	at org.apache.tinkerpop.gremlin.orientdb.gremlintest.OrientGraphProvider.openTestGraph(OrientGraphProvider.java:86)
	at org.apache.tinkerpop.gremlin.AbstractGremlinTest.setup(AbstractGremlinTest.java:83)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.apache.tinkerpop.gremlin.AbstractGremlinSuite.runChild(AbstractGremlinSuite.java:209)
	at org.apache.tinkerpop.gremlin.AbstractGremlinSuite.runChild(AbstractGremlinSuite.java:48)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)

````

File: driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraphFactory.java (hunk 1 above)
Comment (mpollmeier) [inline driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraphFactory.java]: fixed in 28ec888

**Comment by mpollmeier** [comment]
sorry about the delay - looking better now?

**Pull request merged**